docs/workflows/alchemlyb.workflows.base.rst

```diff
@@ -0,0 +1,27 @@
+The base workflow
+=================
+
+The :class:`alchemlyb.workflows.base.WorkflowBase` provides a basic API
+template for the workflow development.
+The workflow should be able to run in an automatic fashion. ::
+
+    >>> from alchemlyb.workflows.base import WorkflowBase
+    >>> workflow = WorkflowBase()
+    >>> workflow.run()
+
+Three main functions are provided such that the workflow could be run in a
+step-by-step fashion. ::
+
+    >>> from alchemlyb.workflows.base import WorkflowBase
+    >>> workflow = WorkflowBase()
+    >>> workflow.read()
+    >>> workflow.preprocess()
+    >>> workflow.estimate()
+    >>> workflow.check_convergence()
+    >>> workflow.plot()
+
```

Comment:
Add more text that describes the conventions that all workflow implementations should follow. For example:

* How are parameters passed? Is the convention that all methods take generic `*args, **kwargs` and then pick out the ones they need? 
* Which attributes are *required* in the class, if any? If it's the one documented in the base class then you can say so and link to the class docs.

Reply:
@orbeckst 
> How are parameters passed? Is the convention that all methods take generic *args, **kwargs and then pick out the ones they need?
Yes, I think "all methods take generic *args, **kwargs and then pick out the ones they need" might be the best initial solution.
>Which attributes are required in the class, if any? If it's the one documented in the base class then you can say so and link to the class docs.
So for the initialization, the creation of the class takes in units, software, T and the output directory. The documentation for this is created automatically by sphinx.

Reply:
Say explicitly what is required in text. Textbooks important to make Intention clear. You’re defining the API here so make it easy for other developers by pointing out what they need to consider. Just „look at existing code“ ist not developer friendly.

Reply:
Also say clearly that the methods will have to take args, kwargs.